apps/testingaccessibility/src/utils/get-active-merchant-coupon.ts

```diff
@@ -41,29 +43,37 @@ export async function getActiveCouponId({
     // compare the discounts if there is a coupon and site/sale running
     incomingCoupon?.merchantCoupon &&
     couponIsValid(incomingCoupon) &&
-    activeDefaultSiteSaleCoupon
+    defaultMerchantCoupon
   ) {
-    const {merchantCoupon} = incomingCoupon
+    const {merchantCoupon: incomingMerchantCoupon} = incomingCoupon
     if (
-      merchantCoupon.percentageDiscount >
-      activeDefaultSiteSaleCoupon.percentageDiscount
+      incomingMerchantCoupon.percentageDiscount >
+      defaultMerchantCoupon.percentageDiscount
     ) {
-      couponId = merchantCoupon.id
+      activeMerchantCoupon = incomingMerchantCoupon
     } else {
-      couponId = activeDefaultSiteSaleCoupon.id
+      activeMerchantCoupon = defaultMerchantCoupon
     }
   } else if (
     // if it's a coupon, use it
     incomingCoupon?.merchantCoupon &&
     couponIsValid(incomingCoupon)
   ) {
-    couponId = incomingCoupon.merchantCoupon.id
+    activeMerchantCoupon = incomingCoupon.merchantCoupon
   } else if (
     // if a sale is running, use that
-    activeDefaultSiteSaleCoupon
+    defaultMerchantCoupon
   ) {
-    couponId = activeDefaultSiteSaleCoupon.id
+    activeMerchantCoupon = defaultMerchantCoupon
   }
 
-  return coupon ? coupon : couponId
+  const defaultCoupon = defaultCoupons?.defaultCoupon
+
+  return {
+    activeMerchantCoupon,
+    ...(defaultCoupon &&
+      defaultCoupon.merchantCouponId === activeMerchantCoupon?.id && {
```

Comment:
only send back the default coupon (with rules) if it matches the active merchant coupon